_posts/2016-04-04-code-smells-part-II.md

```diff
@@ -0,0 +1,91 @@
+
+#### Temporary Field
+
+This case occurs when we are calculating some big algorithm that needs several input variables. Creating these fields in the class has no value most of the times because they are just used for this specific calculation. And this can be dangerous too because you have to be sure you reinitialize them before you start the next computation. 
+Here the best refactoring technique is to use **_Replace Method with Method Object_**, which will extract the method into a separate class. Then you can split the method into several methods within the same class. 
+
+
+#### Refused Bequest
+
+This code smell is a little tricky to detect because this happens when a subclass doesn't use all the behaviour of its parent class. So it's as if the subclass "refuses" some behaviour("bequest") of its parent class. 
+
+In this case, if it makes no sense to continue to have this inheritance, the best refactoring technique is to use **_Delegation_**: we can get rid of the inheritance by creating a field in our subclass and putting in there the parent class. This way every time you need the methods from the parent class just delegate then to this new object. 
+
+When the inheritance is the correct thing to do, then move all unnecessary fields and methods from the subclass. Extract all methods and fields from the subclass and parent class and put them in a new class. Make this new class the SuperClass, from whom the subclass and parent class should inherit. This technique is called **_Extract Superclass_**.
+
+
+#### Alternative Classes with Different Interfaces
+
+Hmm, this case makes me think of "lack of communication" between members of the same team because this happens when we have two classes that make basically the same thing but have different names for their methods. 
+Start by **_Renaming Methods_** or **_Moving Method_**, so you can have both classes implementing the same interface. In some cases, only part of the behaviour is duplicated in both classes. If so, try **_Extract Superclass_** and make the original classes the subclasses.  
+
+
+## Change Preventers
+
+Oh boy! This kind of code smells are the ones you really want to avoid. These are the ones that when you make a change in one place, you have to go basically throughout your code-base making changes in other places too. So it's the nightmare that all of us want to avoid!
+
+
+#### Divergent Change
+
+This is the case when you find yourself changing the same class for several different reasons. This means that you are violating the separation of concerns, i.e the S on [SOLID](https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)) - Single Responsibility Principle.
```

Comment:
violating Single Responsibility Principle (which has to do with separation of concerns)